src/libsyntax/feature_gate.rs

```diff
@@ -1517,6 +1520,16 @@ impl<'a> Visitor<'a> for PostExpansionVisitor<'a> {
             }
         }
 
+        if attr.check_name("macro_export") {
+            if let Some(content) = attr.meta_item_list() {
+                if attr::list_contains_name(&content, "macro_helper_hack") {
+                    gate_feature_post!(&self, macro_helper_hack, attr.span,
```

Comment:
Well, this must be a mistake, the attribute shouldn't be gated at the library side, but rather the effect should be enabled if `feature(use_extern_macros)` is enabled on the use site or stabilized.

Reply:
I was about to write the same thing. This should never happen:

```
error[E0658]: `macro_helper_hack` in `macro_export` is experimental (see issue #35896)
  --> serde-json/src/macros.rs:69:1
   |
69 | #[macro_export(macro_helper_hack)]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add #![feature(macro_helper_hack)] to the crate attributes to enable
```

If not building with `use_extern_macros`, the `macro_helper_hack` should have no effect and should be ignored as it has been on all past stable compilers.